### PR TITLE
[Feature](bangc-ops):add new op rotated_feature_align

### PR DESCRIPTION
**Pull request by tudejiang79**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

add new op rotated_feature_align

## 2. Modification

    1. mlu-ops/bangc-ops/kernels/rotated_feature_align/*
    2. mlu-ops/bangc-ops/mlu_op.h
    3. mlu-ops/bangc-ops/mlu_op_kernel.h
    4. mlu-ops/bangc-ops/kernel_depends.toml
    5. mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/mlu_op_test_proto/*
    6. mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/rotated_feature_align_forward/*
    7. mlu-ops/bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/rotated_feature_align_backward/*
    8. mlu-ops/docs/bangc-docs/user_guide/9_operators/*
    9. mlu-ops/docs/bangc-docs/design_docs/rotated_feature_align/*

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](../docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [x] diff1, diff1 <= 3e-3
- [x] diff2, diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.      |           Details            |      Check Results       |
|----------------|---------------------------|---------------------|
|        1       |          Supported hardware         | MLU270 <br> MLU290 <br>MLU370<br>MLU590|
|        2       |          Job types          |    block <br> U1 <br> U4    |
|        3       |         Layouts            |  NHWC 、NCHW、ARRAY etc    |
|        4       |         Whether multi-dimensions are supported              |                |
|        5       |          Whether element zero is supported             |               |
|        6       |         Data type(half/float)       |         half / float etc           |
|        7      |        Whether there is size limit           |             |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [x] Data type test
- [x] Multidimensional tensor test
- [x] Layout test
- [x] Different size/integer remainder end segment/alignment misalignment test
- [x] Zero dimensional tensor test/zero element test
- [x] stability test
- [x] Multiple platform test
- [x] Gen_case module test
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [x] For memory leak check details, see[GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For code coverage check details, see: [GTest-User-Guide-zh](../docs/GTest-User-Guide-zh.md).
- [x] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

| Test Point         | Acceptance Standard | Test Result (Error Message) |
| -------------- | -------- | -------------------- |
| Whether it conforms to the operator restriction | Normal error |                      |
| Whether illegal parameters are passed  | Normal error |                      |

### 3.2 Performance Test

See [MLU-OPS Performance Acceptance Standard](../docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform ：MLU270

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform ：MLU290

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

Platform：MLU370

|Operator|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-----|----|----|----|----|----|------|-----|
|op_name|   |    |     |    |    |    |     |
|op_name|   |    |     |    |    |    |     |

### 3.3 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

**Review by duzekunKTH — APPROVED** [review]

File: bangc-ops/mlu_op.h [hunk 1]
```diff
@@ -2437,6 +2437,175 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
                      void *grad_input);
 
+// Group:RotatedFeatureAlign
+/*!
+ * @brief Use the feature interpolation to obtain the position information corresponding to the
+ *   refined rotate anchors \b bboxes and reconstructs the feature maps \b output in pixel-wise
+ *   manner to achieve feature alignment.
+ *
+ * @param[in] handle
+ *   Handle to a Cambricon MLUOP context that is used to manage MLU devices and queues in
```
Comment (DanieeelLiu) [inline bangc-ops/mlu_op.h:2447]: handle 前面空格删除，mluops 的规则稍有不同，参考下其他算子的api 注释格式，下面的同样需要检查

File: docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
+# RotatedFeatureAlignBackward 算子开发设计方案
+
+* #### 文档基本信息
+
+| 项目名称      | training v200                                                |
```
Comment (DanieeelLiu) [inline docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md:5]: 项目名称应该不要了吧

File: docs/bangc-docs/design_docs/rotated_feature_align_forward/RotatedFeatureAlignForward.md [hunk 3]
```diff
@@ -0,0 +1,443 @@
+# RotatedFeatureAlignForward 算子开发设计方案
+
+* #### 文档基本信息
+
+| 项目名称      | training v200                                                |
```
Comment (DanieeelLiu) [inline docs/bangc-docs/design_docs/rotated_feature_align_forward/RotatedFeatureAlignForward.md:5]: 同上

File: bangc-ops/mlu_op.h [hunk 4]
```diff
@@ -2437,6 +2437,175 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ *   ::mluOpRotatedFeatureAlignForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] input_desc
+ *   The descriptor of the input tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] input
+ *   Pointer to the MLU memory that stores the input tensor.
+ * @param[in] bboxes_desc
```
Comment (WangYuan-cn) [inline bangc-ops/mlu_op.h:2453]: 格式未修改

File: bangc-ops/mlu_op.h [hunk 5]
```diff
@@ -2437,6 +2437,175 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+
+     param:
+            spatial_scale: 1.0, points: 1
+
+     output array by 1 * 2 * 2 * 1 -->
+         output: [[[[5.0], [6.0]], [[6.0], [8.0]]]]
+     @endverbatim
+ *
+ * @par Reference
+ * - github.com/open-mmlab/mmcv/tree/master/mmcv/ops/rotated_feature_align.py
```
Comment (WangYuan-cn) [inline bangc-ops/mlu_op.h:2517]: 链接补全

File: bangc-ops/mlu_op.h [hunk 6]
```diff
@@ -2437,6 +2437,175 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - The inputs \b bboxes and \b spatial_scale with NaN or infinity are not supported.
+ *
+ * @par Requirements
+ * - None.
+ *
+ * @par Example
+ * - None.
+ *
+ * @par Reference
+ * - github.com/open-mmlab/mmcv/tree/master/mmcv/ops/rotated_feature_align.py
```
Comment (WangYuan-cn) [inline bangc-ops/mlu_op.h:2596]: 链接补全

File: docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md (hunk 2 above)
Comment (WangYuan-cn) [reply docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md:5]: 建议把模板 & 仓库已有文档统一删除。

File: docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+                                                   Tensor bottom_grad)
+```
+
+### 2.2 接口设计
+
+```c++
+mluOpStatus_t MLUOP_WIN_API mluOpRotatedFeatureAlignBackward(const mluOpHandle_t handle,
+                                                          const mluOpTensorDescriptor_t top_output_desc,
+                                                          const void *top_output,
+                                                          const mluOpTensorDescriptor_t bboxes_desc,
```
Comment (WangYuan-cn) [inline docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md:161]: 错位了

File: docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md (hunk 2 above)
Comment (tudejiang79) [reply docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md:5]: 已修改

File: docs/bangc-docs/design_docs/rotated_feature_align_forward/RotatedFeatureAlignForward.md (hunk 3 above)
Comment (tudejiang79) [reply docs/bangc-docs/design_docs/rotated_feature_align_forward/RotatedFeatureAlignForward.md:5]: 已修改

File: bangc-ops/mlu_op.h (hunk 4 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2453]: 已修改

File: bangc-ops/mlu_op.h (hunk 6 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2596]: 已修改

File: docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md (hunk 7 above)
Comment (tudejiang79) [reply docs/bangc-docs/design_docs/rotated_feature_align_backward/RotatedFeatureAlignBackward.md:161]: 已修改

**Review by DanieeelLiu — APPROVED** [review]

File: bangc-ops/mlu_op.h [hunk 8]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
                      const mluOpTensorDescriptor_t grad_input_desc,
                      void *grad_input);
 
+// Group:RotatedFeatureAlign
+/*!
+ * @brief Use the feature interpolation to obtain the position information corresponding to the
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2660]: Use > Uses

File: bangc-ops/mlu_op.h [hunk 9]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ *
+ * @param[in] handle
+ * Handle to a Cambricon MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRotatedFeatureAlignForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] input_desc
+ * The descriptor of the input tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] input
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] bboxes_desc
+ * The descriptor of bboxes, containing the dimension and layout of bboxes tensor.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2672]: The descriptor of bboxes, which contains the dimension and layout of bboxes tensor.

File: bangc-ops/mlu_op.h [hunk 10]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] bboxes
+ * Pointer to the MLU memory that stores the bboxes tensor.
+ * @param[in] spatial_scale
+ * A float value that is the scale factor of coordinates of bboxes.
+ * @param[in] points
+ * An int value that is the number of sample points. Only 1 and 5 are supported. Defaults to 1.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2679]: Defaults to 1. > The default value is 1.

File: bangc-ops/mlu_op.h [hunk 11]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * @param[in] output_desc
+ * The descriptor of output tensor, containing the dimension and layout of output tensor.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2681]: The descriptor of output tensor, which contains the dimension and layout of output tensor.

File: bangc-ops/mlu_op.h [hunk 12]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * @param[out] output
+ * Pointer to the MLU memory that stores the output tensor.
+ * 
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM, ::MLUOP_STATUS_NOT_SUPPORTED.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2686]: 句尾句号去掉

File: bangc-ops/mlu_op.h [hunk 13]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - The supported data layouts of \b input, \b bboxes and \b output are as follows:
+ *   - input tensor: \p MLUOP_LAYOUT_NHWC.
+ *   - bboxes tensor: \p MLUOP_LAYOUT_ARRAY.
+ *   - output tensor: \p MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The input tensor and output tensor must be 4D.
+ * - Size of the each dimension of input tensor and output tensor must be the same.
+ * - The bboxes tensor must be 4D.
+ * - The information of the fourth dimension of bboxes tensor consists of [y, x, w, h, a].
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2705]: fourth dimension 》4D

File: bangc-ops/mlu_op.h [hunk 14]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the first dimension of input tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2706]: The 1D sizes of input tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 15]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the second dimension of input tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2707]: The 2D sizes of input tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 16]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the third dimension of input tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2708]: The 3D sizes of input tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 17]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - The shape of \b input should be [batch_num, height, width, channels].
+ * - The shape of \b bboxes should be [batch_num, height, width, 5].
+ * - The shape of \b output should be [batch_num, height, width, channels].
+ * - \b points is the number of sample points. Only 1 and 5 are supported. Defaults to 1.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2712]: The default value is 1.

File: bangc-ops/mlu_op.h [hunk 18]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ *
+ * @param[in] handle
+ * Handle to a Cambricon MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRotatedFeatureAlignBackward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] top_output_desc
+ * The descriptor of the top_output tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] top_output
+ * Pointer to the MLU memory that stores the top_output tensor.
+ * @param[in] bboxes_desc
+ * The descriptor of bboxes, containing the dimension and layout of bboxes tensor. For detailed
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2762]: The descriptor of bboxes, which contains the dimension and layout of bboxes tensor.

File: bangc-ops/mlu_op.h [hunk 19]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * Pointer to the MLU memory that stores the top_output tensor.
+ * @param[in] bboxes_desc
+ * The descriptor of bboxes, containing the dimension and layout of bboxes tensor. For detailed 
+ * information, see ::mluOpTensorDescriptor_t.
+ * @param[in] bboxes
+ * Pointer to the MLU memory that stores the bboxes tensor.
+ * @param[in] spatial_scale
+ * A float value that is the scale factor of coordinates of bboxes.
+ * @param[in] points
+ * An int value that is the number of sample points. Only 1 and 5 are supported. Defaults to 1.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2769]: The default value is 1.

File: bangc-ops/mlu_op.h [hunk 20]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * @param[in] bottom_input_desc
+ * The descriptor of bottom_input tensor, containing the dimension and layout of bottom_input tensor.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2771]: containing > which contains

File: bangc-ops/mlu_op.h [hunk 21]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * @param[out] bottom_input
+ * Pointer to the MLU memory that stores the bottom_input tensor.
+ * 
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM, ::MLUOP_STATUS_NOT_SUPPORTED.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2776]: 句尾句号去掉

File: bangc-ops/mlu_op.h [hunk 22]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ *
+ * @par Data Layout
+ * - The supported data layouts of \b top_output, \b bboxes and \b bottom_input are as follows:
+ *   - top_output tensor: \p MLUOP_LAYOUT_NHWC.
+ *   - bboxes tensor: \p MLUOP_LAYOUT_ARRAY.
+ *   - bottom_input tensor: \p MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The top_output tensor and bottom_input tensor must be 4D.
+ * - Size of the each dimension of top_output tensor and bottom_input tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2794]: Sizes

File: bangc-ops/mlu_op.h [hunk 23]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - The bboxes tensor must be 4D.
+ * - The information of the fourth dimension of bboxes tensor consists of [y, x, w, h, a].
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2796]: The 4D information of bboxes tensor consists of [y, x, w, h, a].

File: bangc-ops/mlu_op.h [hunk 24]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the first dimension of top_output tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2797]: The 1D sizes of top_output tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 25]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the second dimension of top_output tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2798]: The 2D sizes of top_output tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 26]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - Size of the third dimension of top_output tensor and bboxes tensor must be the same.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2799]: The 3D sizes of top_output tensor and bboxes tensor must be the same.

File: bangc-ops/mlu_op.h [hunk 27]
```diff
@@ -2655,6 +2655,177 @@ mluOpRoiCropBackward(mluOpHandle_t handle,
...
+ * - The shape of \b top_output should be [batch_num, height, width, channels].
+ * - The shape of \b bboxes should be [batch_num, height, width, 5].
+ * - The shape of \b bottom_input should be [batch_num, height, width, channels].
+ * - \b points is the number of sample points. Only 1 and 5 are supported. Defaults to 1.
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:2803]: The default value is  1.

**Review by GraceGuanH — COMMENTED** [review]
Please check the comments.

File: bangc-ops/mlu_op.h (hunk 8 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2660]: 已修改

File: bangc-ops/mlu_op.h (hunk 9 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h]: 已修改

File: bangc-ops/mlu_op.h (hunk 10 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2679]: 已修改

File: bangc-ops/mlu_op.h (hunk 11 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h]: 已修改

File: bangc-ops/mlu_op.h (hunk 12 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2686]: 已修改

File: bangc-ops/mlu_op.h (hunk 13 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2705]: 已修改

File: bangc-ops/mlu_op.h (hunk 14 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2706]: 已修改

File: bangc-ops/mlu_op.h (hunk 15 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2707]: 已修改

File: bangc-ops/mlu_op.h (hunk 16 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2708]: 已修改

File: bangc-ops/mlu_op.h (hunk 17 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2712]: 已修改

File: bangc-ops/mlu_op.h (hunk 18 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h]: 已修改

File: bangc-ops/mlu_op.h (hunk 19 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2769]: 已修改

File: bangc-ops/mlu_op.h (hunk 20 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h]: 已修改

File: bangc-ops/mlu_op.h (hunk 21 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2776]: 已修改

File: bangc-ops/mlu_op.h (hunk 22 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2794]: 已修改

File: bangc-ops/mlu_op.h (hunk 23 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2796]: 已修改

File: bangc-ops/mlu_op.h (hunk 24 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2797]: 已修改

File: bangc-ops/mlu_op.h (hunk 25 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2798]: 已修改

File: bangc-ops/mlu_op.h (hunk 26 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2799]: 已修改

File: bangc-ops/mlu_op.h (hunk 27 above)
Comment (tudejiang79) [reply bangc-ops/mlu_op.h:2803]: 已修改

**Review by WangYuan-cn — APPROVED** [review]

**Review by GraceGuanH — APPROVED** [review]

**Pull request merged**